File: backend/src/main/java/com/backend/backend/message/Message.java
package com.backend.backend.message;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "Messages")
public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "toId")
    private long toId;

    @Column(name = "fromId")
    private long fromId;

    @Column(name = "content")
    private String content;

    @Column(name = "isRead")
    private Boolean isRead;

    @Column(name = "timeStamp")
    private long timeStamp;

    @Column(name = "chatroomId")
    private long chatroomId;
}

File: fightme_webapp/lib/Models/message.dart
class Message {
  int fromId = 0;
  int toId = 0;
  int chatroomId = 0;
  String content = "";
  bool isRead = false;
  int timeStamp = 0;

  Message.fromJson(Map<String, dynamic> json) {
    fromId = json['fromId'];
    toId = json['toId'];
    content = json['content'];
    isRead = json['isRead'];
    timeStamp = json['timeStamp'];
    chatroomId = json['chatroomId'];
  }

  Map<String, dynamic> toJson() => {
        'toId': toId,
        'fromId': fromId,
        'content': content,
        'isRead': false,
        'chatroomId': chatroomId,
        'timeStamp': timeStamp
      };

  Message(int to, int from, String text, int chatId) {
    toId = to;
    fromId = from;
    content = text;
    isRead = false;
    timeStamp = 0;
    chatroomId = chatId;
  }
}
